scipy/stats/stats.py

```diff
@@ -3565,6 +3568,159 @@ def count_rank_tie(ranks):
     # Limit range to fix computational errors
     return KendalltauResult(min(1., max(-1., tau)), pvalue)
 
+WeightedTauResult = namedtuple('WeightedTauResult', ('correlation', 'pvalue'))
+
+
+def weightedtau(x, y, rank=True, weigher=lambda x: 1./(x+1), additive=True):
```

Comment:
I haven't benchmarked this, but it looks like passing a Python function to the inner loops of your Cython code will be quite inefficient. Does it make sense to at least make the default fast by using ``weigher=None`` and implementing the default ``x./(x+1)`` on the Cython side?